codes/c/chapter_backtracking/n_queens.c

```diff
@@ -0,0 +1,89 @@
+/**
+ * File: n_queens.c
+ * Created Time: 2023-09-25
+ * Author: lucas (superrat6@gmail.com)
+ */
+
+#include "../utils/common.h"
+
+#define MAX_N 100
+#define MAX_SOLUTIONS 1000
+
+char ***results;
+int resultsCount = 0;
```

Comment:
Removed here and created in `nQueens` function
Try not to use global variables

Here I think it would be better to create a `Result` struct

Reply:
Such as
```c
struct result {
    char ***results;
    int resultsCount;
};


typedef struct result Result;
```

Reply:
I think using the commit `39fdacacd67875030bf5b05898faebee48d75c9` is better

Reply:
> I think using the commit `39fdacacd67875030bf5b05898faebee48d75c9` is better

Yes, the code of `39fdacacd67875030bf5b05898faebee48d75c9` is very good in practice
However, parameter signatures are inconsistent compared with other languages, which will increase reading difficulties.

Reply:
@Gonglja Done.

Reply:
> @Gonglja Done.

Thanks @lhxsm, your changes have been received. The format of the code submitted this time has changed significantly from the previous one and needs to be changed to the previous format.